.github/workflows/rust.yml

```diff
@@ -0,0 +1,94 @@
+name: Rust
+
+on: [push, pull_request]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  check_n_test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+          - macos-latest
+        toolchain:
+          - stable
+          - nightly
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v3
+
+      - name: Install toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: ${{ matrix.toolchain }}
+          override: true
+
+      - name: Install dependencies
+        if: ${{ matrix.os == 'ubuntu-latest' }}
+        run: sudo apt update && sudo apt install libomp-dev
+
+      - name: Install dependencies
+        if: ${{ matrix.os == 'macos-latest' }}
+        run: brew install llvm libomp cmake
+
+      - name: Run cargo check
+        working-directory: "./barretenberg_wrapper"
+        run: |
+          cargo +${{ matrix.toolchain }} check --all-targets --verbose
+
+      # TODO: Enable when tests work
+      # - name: Run cargo test
+      #   working-directory: "./barretenberg_wrapper"
+      #   run: |
+      #     cargo +${{ matrix.toolchain }} test
```

Comment:
On linux, running `cargo test` fails with a ton of linker errors. I'm not sure why.

On Mac, the tests pedersen tests fail with (what looks like) an ordering problem.

I'd love some help getting the tests working in this project.

Reply:
I've updated the tests to match `aztec_backend` as per @vezenovm's suggestions and re-enabled the test step. I'm guessing we'll still have issues with linking on Linux though.

Reply:
The linux tests were failing because rust picks up gcc as the linker on ubuntu. I've added a config file to override that as per https://github.com/rust-lang/rust/issues/71515#issuecomment-935020057

Reply:
Also, it seems by specifying this inside of barretenberg_wrapper, it'll apply when used as a dependency as per the hierarchical-structure documentation: https://doc.rust-lang.org/cargo/reference/config.html#hierarchical-structure